src/irmin-pack/inode.ml

```diff
@@ -1414,13 +1444,42 @@ struct
   type key = Key.t
   type value = Inter.Val.t
 
+  exception Invalid_depth = Inter.Raw.Invalid_depth
+
   let mem t k = Pack.mem t k
+  let pp_value = Irmin.Type.pp Inter.Raw.t
+
+  let err_invalid_depth expected got v =
+    raise (Invalid_depth { expected; got; v })
+
+  let pp_invalid_depth ppf (expected, got, v) =
+    Fmt.pf ppf "Invalid depth: got %d, expecting %d (%a)" got expected pp_value
+      v
+
+  let check_depth v e g = if g <> e then err_invalid_depth e g v
+
+  (* call be recursive inners calls to find *)
+  let check_depth_inner ~init_depth ~depth v =
+    match v with
+    | None -> ()
+    | Some v -> (
+        match (init_depth, Inter.Raw.depth v) with
+        | None, _ ->
+            (* recursive calls only happens on non-value root nodes. *)
+            assert false
+        | Some d0, Some d -> check_depth v depth (d0 + d)
```

Comment:
When `Inter.Raw.depth` returns `Some d`, `d` is the depth from the root inode, so the check should be `check_depth v depth d` (no addition).

I think it runs fine because `d0` is always equal to `0`